packages/components/src/components/SquiggleViewer/VariableBox.tsx

```diff
@@ -101,57 +132,95 @@ export const VariableBox: FC<VariableBoxProps> = ({
     };
   }, []);
 
+  const isCollapsed = isFocused ? false : settings.collapsed;
+  const shouldShowTriangleToggle = !isFocused;
+  const shouldShowLeftHeaderPreview = !isFocused && !!preview;
+  const shouldShowRightHeaderItems = !settings.collapsed || isFocused;
+  const shouldShowRightHeaderString = Boolean(
+    shouldShowRightHeaderItems && path.items.length
+  );
+  const shouldShowRightHeaderFindInEditor = !isRoot;
+  const showOpenedLeftSidebar = Boolean(
+    !isCollapsed && path.items.length && !isFocused
+  );
```

Comment:
Opinionated note on coding style; feel free to ignore if you disagree, but I think it'd be interesting to explicate.

There are two consequences of extracting small one-use variables and functions outside of long expressions:
- first, you give them names (good)
- second, when you read or refactor the code, you have to jump to definition more often (bad)

In our old Rescript styleguide, we had a rule (which wasn't strictly followed anyway) about something like "extract all pattern matching that are longer than 3 lines in functions", that I also thought was excessive.

Extracting boolean conditions like that can often make code less readable, in my experience.

In some cases, it would also block you from using typescript guards; for example:

```
function printDoubleXIfDefined(x: number | undefined) {
  // this won't work:
  const print = x !== undefined;
  if (print) {
    console.log(x * 2); // error, x might be undefined
  }

  // this is fine:
  if (x !== undefined) {
    console.log(x * 2); // typescript automatically narrows `x` type in `if` body
  }
}
```

This only applies to one-use vars, of course, and I can see the benefit of extracting bool vars if some of them depend on others (but in the code above, there's only one dependency).

One way to document what's going on and keep the conditions inlined is to add comments instead of extracting vars:
```
// show opened left sidebar?
Boolean(!isCollapsed && path.items.length && !isFocused) && (
  <...>
)
```

Reply:
This is similar to how React violated the classical separation of concerns (HTML/CSS/JS) in favor of component-level organization.

When the code gets complicated enough it's useful to split it into several components, but in that case the component and the conditions for rendering that component should still stay close to each other. (I'm not sure how hard it would be in this case, there's a lot going on here and there are many shared vars that might make it difficult).

Reply:
Good point, I'll think through this more. I generally agree with you - however, I found this specific component so complex that it seemed extra useful to split out this logic to be in one place.